transport.go

```diff
@@ -190,51 +204,87 @@ func (t *HTTPTransport) SendEvent(event *Event) {
 			t.dsn.projectID,
 		)
 	default:
-		t.wg.Done()
 		Logger.Println("Event dropped due to transport buffer being full.")
-		// worker would block, drop the packet
 	}
+
+	t.buffer <- b
 }
 
 // Flush notifies when all the buffered events have been sent by returning `true`
 // or `false` if timeout was reached.
 func (t *HTTPTransport) Flush(timeout time.Duration) bool {
-	c := make(chan struct{})
+	toolate := time.After(timeout)
+
+	var b batch
+	for {
+		// Wait until processing the current batch has started or the timeout.
```

Comment:
We must wait until the `worker` has seen the current batch, because it is the only way `b.done` will be closed.
If we do not wait, there is a possible execution flow in which `b.done` is never closed, and the only way out of Flush would be waiting for the time out.